src/fortran/client.F90

```diff
@@ -139,32 +139,45 @@ module smartredis_client
 contains
 
 !> Initializes a new instance of a SmartRedis client
-subroutine initialize_client( self, cluster )
+function initialize_client(self, cluster)
+  integer(kind=enum_kind)           :: initialize_client
   class(client_type), intent(inout) :: self    !< Receives the initialized client
   logical, optional,  intent(in   ) :: cluster !< If true, client uses a database cluster (Default: .false.)
 
+  ! Local variables
+  integer(kind=enum_kind) :: result
+
   if (present(cluster)) self%cluster = cluster
-  self%client_ptr = c_constructor(self%cluster)
-  self%is_initialized = .true. 
-end subroutine initialize_client
+  result = c_constructor(self%cluster, self%client_ptr)
+  self%is_initialized = result .eq. SRNoError
+  initialize_client = result
+end function initialize_client
 
+!> Check whether the client has been initialized
 logical function isinitialized(this)
   class(client_type) :: this
   isinitialized = this%is_initialized
 end function isinitialized
 
 !> A destructor for the SmartRedis client
-subroutine destructor( self )
+function destructor(self)
+  integer(kind=enum_kind)           :: destructor
   class(client_type), intent(inout) :: self
 
-  call c_destructor(self%client_ptr)
+  ! Local variables
+  integer(kind=enum_kind) :: result
+
+  result = c_destructor(self%client_ptr)
   self%client_ptr = C_NULL_PTR
-end subroutine destructor
+  destructor = result
+end function destructor
 
 !> Check if the specified key exists in the database
-logical function key_exists(self, key)
-  class(client_type), intent(in) :: self !< The client
-  character(len=*),   intent(in) :: key  !< The key to check
+function key_exists(self, key, exists)
```

Comment:
Any particular reason why you moved the declaration of the function output type to the argument list below? If you're going to keep this convention, I might suggest that you instead use the following construct so that you don't get multiple repeated names for functions and variables in quick succession:
```
function foo( bar ) result (baz)
  integer, intent(in   ) :: bar
  integer, intent(  out) :: baz
end function foo
```

Reply:
The length of the type declaration was starting to push too much of the function prototype into the next line, so that's why I broke it up. I've adopted your suggestion except that my compiler is complaining when I put an intent on the output variable.